viewing_party/party.py

```diff
@@ -1,23 +1,360 @@
+    for movie in user_data[category]:
+        movies_list.append(movie["title"])
+
+    return movies_list
+
+def get_friends_watched_movies_list(user_data):
+    '''
+    helper function get_friends_watched_movies_list
+    - creates a list of titles of a "watched" category
+    - returns list of titles
+    '''
+    #creating an empty list
+    movies_list = []
+    #using for loop to iterate by each friend
+    for friend in user_data["friends"]:
+        #using for loop to iterate by each movie in friend's watched list
+        for movie in friend["watched"]:
+            #append all the titles to the list
+            movies_list.append(movie["title"])
+    return movies_list
+
+def use_set_to_compare_lists_return_list(main_list, list_to_compare):
+    '''
+    helper function use_set_to_compare_lists_return_list
+    - takes to lists (list 1, list 2)
+    - returns the difference between list 1 and list 2, 
+    - unique values of a list 1, that are not found in the list 2
+    '''
+    #convirting lists we received in the function call to sets
+    main_set = set(main_list)
+    set_to_compare = set(list_to_compare)
+
+    #calculating set difference
+    difference_set = main_set - set_to_compare
+    #re-convirting this difference set to list
+    difference_list = list(difference_set)
+
+    return difference_list
+
+def convert_titles_into_list_of_movies_dict(user_data, list_of_titles, user, category):
+    '''
+    helpers function convert_titles_into_list_of_movies_dict
+    - takes user_data, list_of_titles, user and category
+    - user will be True for user and False for friends
+    - category might be "watched", "watchlist" or "favorites"
+    - uses user_data and list_of_titles to create a movie dictionary with all the movie info inside
+    - returns the list "result" with movie dictionaries
+    '''
+
+    #empty result list
+    result = []
+    #using for loop to iterate through our titles in list_of_titles
+    for title in list_of_titles:
+        # if user is True, which means that we are working with user info
+        if user:
+            #using for loop to iterate through each movie in a category
+            for movie in user_data[category]:
+                # and if we find our title,
+                if title == movie["title"]:
+                    # we add the whole movie to our result list
+                    result.append(movie)
+        else: #if user is false, we're working with friends info
+            #for each friend in the list
+            for friend in user_data["friends"]:
+                #for each movie in "watched"
+                for movie in friend["watched"]:
+                    # and if we find our title and movie is not yet there from another friend,
+                    if title == movie["title"] and movie not in result:
+                        # we add the whole movie to our result list
+                        result.append(movie)
+
+    return result

```

Comment:
I'm very intrigued by this helper function and honestly very impressed with all your helper functions already! My only suggestion is that you distill everything down to lists of titles. This is ok, but it does augment the code quite a bit. If you were able to find a way to convert all your other helper functions so they are working with the entire movies themselves instead of just the title, you wouldn't even need this function at all!.